com.unity.render-pipelines.universal/ShaderLibrary/SSAO.hlsl

```diff
@@ -0,0 +1,251 @@
+#ifndef UNIVERSAL_SSAO_INCLUDED
+#define UNIVERSAL_SSAO_INCLUDED
+
+// Includes
+#include "Packages/com.unity.render-pipelines.core/ShaderLibrary/Common.hlsl"
+#include "Packages/com.unity.render-pipelines.universal/ShaderLibrary/DeclareDepthTexture.hlsl"
+#include "Packages/com.unity.shadergraph/ShaderGraphLibrary/ShaderVariablesFunctions.hlsl"
+//#include "Packages/com.unity.render-pipelines.universal/ShaderLibrary/DeclareNormalsTexture.hlsl"
+
+// Textures & Samplers
+TEXTURE2D_X(_BaseMap);
+TEXTURE2D_X(_ScreenSpaceOcclusionTexture);
+
+SAMPLER(sampler_BaseMap);
+SAMPLER(sampler_ScreenSpaceOcclusionTexture);
+
+// Params
+float4 _BlurOffset;
+float4 _SSAOParams;
+float4 _BaseMap_TexelSize;
+float4 _CameraDepthTexture_TexelSize;
+
+// SSAO Settings
+#define DOWNSAMPLE _SSAOParams.x
+#define INTENSITY _SSAOParams.y
+#define RADIUS _SSAOParams.z
+
+// GLES2: In many cases, dynamic looping is not supported.
+#if defined(SHADER_API_GLES) && !defined(SHADER_API_GLES3)
+    #define SAMPLE_COUNT 3
+#else
+    #define SAMPLE_COUNT _SSAOParams.w
+#endif
+
+// Function defines
+#define SCREEN_PARAMS        GetScaledScreenParams()
+#define SAMPLE_BASEMAP(uv)   SAMPLE_TEXTURE2D_X(_BaseMap, sampler_BaseMap, UnityStereoTransformScreenSpaceTex(uv));
+#define SAMPLE_BASEMAP_R(uv) SAMPLE_TEXTURE2D_X(_BaseMap, sampler_BaseMap, UnityStereoTransformScreenSpaceTex(uv)).r;
+
+// Constants
+// kContrast determines the contrast of occlusion. This allows users to control over/under
+// occlusion. At the moment, this is not exposed to the editor because it's rarely useful.
+static const float kContrast = 0.6;
+
+// The constants below are used in the AO estimator. Beta is mainly used for suppressing
+// self-shadowing noise, and Epsilon is used to prevent calculation underflow. See the
+// paper (Morgan 2011 http://goo.gl/2iz3P) for further details of these constants.
+static const float kBeta = 0.002;
+#define EPSILON         1.0e-4
+
+
+float2 GetScreenSpacePosition(float2 uv)
+{
+    return uv * SCREEN_PARAMS.xy * DOWNSAMPLE;
+}
+
+// Sample point picker
+float3 PickSamplePoint(float2 uv, float randAddon, int index)
+{
+    float2 positionSS = GetScreenSpacePosition(uv);
+    float noise = InterleavedGradientNoise(positionSS, index);
+    return SampleSphereUniform(frac(noise), (noise * PI));
+}
+
+float SampleAndGetLinearDepth(float2 uv)
+{
+    float rawDepth = SampleSceneDepth(uv.xy).r;
+    #if defined(_ORTHOGRAPHIC)
+        float linearDepth = ((_ProjectionParams.z - _ProjectionParams.y) * (1.0 - rawDepth) + _ProjectionParams.y);
+    #else
+        float linearDepth = LinearEyeDepth(rawDepth, _ZBufferParams);
```

Comment:
I wonder if we should add support to handle ortographic case in LinearEyeDepth (not for this PR)

Reply:
Yes we definitely should!